refactor : enhance icon size and change component name

ElsayedMagdy122 opened the pull request:
- replace box with column 
- change component name
- enhance icon size
- fix : #25 
![image](https://github.com/user-attachments/assets/58c70139-6f4e-43ed-a5c0-7306dbfd1646)

ElsayedMagdy122 committed:
refactor : enhance icon size and chane component name

ElsayedMagdy122 committed:
Merge remote-tracking branch 'origin/common-composables' into component/bottom-nav-bar-refactored

Al-Qassim commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -30,19 +30,20 @@ import com.cairosquad.design_system.R
     selectedItemIndex: Int = 0,
 ) {
-    Box(
+    Column(
```
you should not use a box, column, or a row, if contains only one sub-component

just apply it's modifier to the row inside it, and remove this column

![image](https://github.com/user-attachments/assets/475978ce-ac5e-46b1-9539-b0f9c4f7c786)

ElsayedMagdy122 replied:
my friend it is not problem

Al-Qassim commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -30,19 +30,20 @@ import com.cairosquad.design_system.R
     navigationItems: List<BottomNavItem>,
     onItemSelected: (Int) -> Unit,
     modifier: Modifier = Modifier,
     selectedItemIndex: Int = 0,
```
I think `selectedItemIndex` is better to not be an optional parameter, the value 0 not spacial, it's better to always let the user specify this value

ElsayedMagdy122 replied:
@Al-Qassim no , must contain start destenation by default home that index 0

gehadepada replied:
agreed with qassim ,selectedItemIndex it will based on user choice ,when you navigate to this screen selectedItemIndex value will passed

ElsayedMagdy122 replied:
@gehadepada There should be a start screen that is displayed when the user opens the application, which is the home screen.

Al-Qassim commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -30,19 +30,20 @@ import com.cairosquad.design_system.R
 @Composable
-fun MovioNavigationBar(
+fun NavigationBar(
     navigationItems: List<BottomNavItem>,
```
I recommend making the navigationItems an optional parameter, with a default value of the items in the current design, so that if the user of this component want to use this component, he will not need to define this list and pass it every time for the current design, also this will not prevent the user from customizing the nav bar, he can always make another list and pass it. do you think this is a bad idea?

![image](https://github.com/user-attachments/assets/49edd2d5-d8bf-41a8-b473-8434f43cee99)

YoussefmSaber replied:
This is a bad approach Qassim, why is that, if we made a default navigation items, then how can the developer know that he has to provide his list of items.
how about in the feuture we wanted to add a new navigation item, so we will have to make a new list with the same data and add the new item to the new list then provide it to the navigationBar component, instead of adding a new item to the list that we pass to the NavigationBar.

gehadepada replied:
we can make the list as  public constant or variable to avoid recreating of the list @YoussefmSaber 
then why he should create new list ??

Al-Qassim commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -52,25 +53,25 @@ fun MovioNavigationBar(
                     )
                     Text(
-                        modifier = Modifier.align(Alignment.BottomCenter),
                         text = item.label,
```
for the `BottomNavItem` data class, it's better to set the label as a string resource int, it will be easier to use and consistent with the other parameters

```
data class BottomNavItem(
    @DrawableRes val unColoredIcon: Int,
    @DrawableRes val coloredIcon: Int,
    val label: String // replace with @StringRes val label: Int
)
```

at this line, and line 69, you can use
`text = stringResource(item.label)`

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -30,19 +30,20 @@ import com.cairosquad.design_system.R
         modifier = modifier
             .background(color = Theme.color.surfaces.surface)
             .windowInsetsPadding(WindowInsets.navigationBars)
+            .padding(horizontal = 20.dp)
```
why you deleted vertical padding ??

gehadepada commented on design_system/src/main/java/com/cairosquad/design_system/component/NavigationBar.kt:
```diff
@@ -52,25 +53,25 @@ fun MovioNavigationBar(
+                    if (selectedItemIndex == index) Theme.color.brand.primary else Theme.color.surfaces.onSurfaceVariant
+                Column(
                     modifier = Modifier
                         .size(43.dp)
```
you give it size of 43 dp but width in figma =74

yasserahmed10 committed:
refactor: update project dependencies to use domain module

yasserahmed10 committed:
refactor: rename usecase .gitignore to project root

yasserahmed10 committed:
feat: integrate new modules into app build

Adds `:domain`, `:entity`, `:remote`, `:repository`, `:ui`, and `:viewmodel` modules as dependencies to the main application module.

yasserahmed10 committed:
feat: add Koin dependency and KSP configuration

yasserahmed10 committed:
feat: implement Koin dependency injection in application

yasserahmed10 committed:
feat: set application class in AndroidManifest.xml

yasserahmed10 committed:
Merge pull request #32 from Cairo-Squad/feature/add-koin

Feature/add koin

ElsayedMagdy122 committed:
Merge remote-tracking branch 'origin/develop' into component/bottom-nav-bar-refactored

# Conflicts:
#	gradle/libs.versions.toml

ElsayedMagdy122 committed:
refactor : remove unnecessary file

ElsayedMagdy122 committed:
refactor :add navigation dependency

ElsayedMagdy122 committed:
refactor : enhance implementation
